sync/src/main/java/org/bf2/sync/SecretManager.java

```diff
@@ -0,0 +1,211 @@
+                .withNewTls()
+                    .withNewCertRef(ENDPOINT_TLS_CRT, masterSecretName, null)
+                    .withNewKeyRef(ENDPOINT_TLS_KEY, masterSecretName, null)
+                .endTls()
+                .endEndpoint();
+        }
+
+        if (isKafkaAuthenticationEnabled(managedKafka)) {
+            replacement.editOauth()
+                    .withNewClientIdRef(OAUTH_SSO_CLIENT_ID, masterSecretName, null)
+                    .withClientId(null)
+                    .withNewClientSecretRef(OAUTH_SSO_CLIENT_SECRET, masterSecretName, null)
+                    .withClientSecret(null)
+                .endOauth();
+        }
+
+        managedKafka.getServiceAccount(ServiceAccount.ServiceAccountName.Canary)
+            .ifPresent(serviceAccount -> {
+                replacement.editMatchingServiceAccount(sa -> serviceAccount.getName().equals(sa.getName()))
+                    .withNewPrincipalRef(CANARY_SASL_PRINCIPAL, masterSecretName, null)
+                    .withPrincipal(null)
+                    .withNewPasswordRef(CANARY_SASL_PASSWORD, masterSecretName, null)
+                    .withPassword(null)
+                .endServiceAccount();
+            });
+
+       return new ManagedKafkaBuilder(managedKafka)
+                .withSpec(replacement.build())
+                .build();
+    }
+
+    public void calculateMasterSecretDigest(ManagedKafka managedKafka, Secret secret) {
+        var metadata = new ObjectMetaBuilder(managedKafka.getMetadata())
+                .addToAnnotations(ANNOTATION_MASTER_SECRET_DIGEST, buildDigest(decode(secret.getData())))
+                .build();
+
+        managedKafka.setMetadata(metadata);
+    }
+
+    private static String encode(String value) {
+        return Base64.getEncoder().encodeToString(value.getBytes(StandardCharsets.UTF_8));
+    }
+
+    private static String decode(String value){
+        return new String(Base64.getDecoder().decode(value), StandardCharsets.UTF_8);
+    }
+
+    private static Map<String, String> decode(Map<String, String> data){
+        return data.entrySet()
+                .stream()
+                .map(entry -> Map.entry(entry.getKey(), decode(entry.getValue())))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private static String buildDigest(Map<String, String> data) {
+        final MessageDigest secretsDigest;
+
+        try {
+            secretsDigest = MessageDigest.getInstance("SHA-1");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+
+        // Sort by key to ensure iteration order for any input map
+        data.entrySet()
+            .stream()
+            .sorted(Comparator.comparing(Map.Entry::getKey))
+            .forEach(entry ->
+                secretsDigest.update(entry.getValue().getBytes(StandardCharsets.UTF_8)));
+
+        return String.format("%040x", new BigInteger(1, secretsDigest.digest()));
```

Comment:
would it be possible to extract some common secret logic between this and SecuritySecretManager?